pkg/controller/datadogagentdeployment/datadogagentdeployment_controller.go

```diff
@@ -162,6 +162,15 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	// Watch for changes to secondary resource Endpoints and requeue the owner DatadogAgentDeployment
+	err = c.Watch(&source.Kind{Type: &corev1.Endpoints{}}, &handler.EnqueueRequestForOwner{
```

Comment:
would it be possible to watch Endpoints only when the Cluster Agent is enabled?

Reply:
it is not trivial to do, but it could be done.